src/pages-and-resources/discussions/app-config-form/apps/shared/discussion-topics/TopicItem.jsx

```diff
@@ -0,0 +1,159 @@
+}) => {
+  const [title, setTitle] = useState(name);
+  const [isRemove, setIsRemove] = useState(false);
+  const {
+    handleChange,
+    handleBlur,
+    touched,
+    errors,
+  } = useFormikContext();
+
+  useEffect(() => {
+    setTitle(name);
+  }, [name]);
+
+  const isInvalidtopicNameKey = (
+    (touched?.discussionTopics && touched.discussionTopics[index]?.name)
+    && (errors?.discussionTopics && errors?.discussionTopics[index]?.name)
+  );
+
+  const isExistedName = (
+    (touched?.discussionTopics && touched.discussionTopics[index]?.name)
+    && (errors && errors[index]?.name)
+  );
+
+  const getHeading = (isOpen = false) => {
+    let heading;
+    if (!title) {
+      heading = <span className="h4 py-2 mr-auto">Configure topic</span>;
+    } else if (isOpen) {
+      heading = <span className="h4 py-2 mr-auto">Rename {title} topic</span>;
+    } else {
+      heading = <span className="py-2">{title}</span>;
+    }
+    return heading;
+  };
+
+  const handleToggle = (isOpen) => {
+    if (!isOpen && !isInvalidtopicNameKey) {
+      setTitle(name);
+    }
+  };
+
+  const DeleteDiscussionTopic = (event) => {
+    event.stopPropagation();
+    setIsRemove(true);
+  };
+
+  return (
+    <>
+      {
+        isRemove ? (
+          <Card className="rounded mb-3 p-1">
+            <Card.Body>
+              <div className="text-primary-500 mb-2 h4">
+                {intl.formatMessage(messages.discussionTopicDeletionLabel)}
+              </div>
+              <Card.Text className="text-justify text-muted">
+                {intl.formatMessage(messages.discussionTopicDeletionHelp)}
+              </Card.Text>
+              <div className="d-flex justify-content-end">
+                <Button
+                  variant="tertiary"
+                  onClick={() => setIsRemove(false)}
+                >
+                  {intl.formatMessage(messages.cancelButton)}
+                </Button>
+                <Button
+                  variant="outline-brand"
+                  className="ml-2"
+                  onClick={() => onDelete()}
+                >
+                  {intl.formatMessage(messages.deleteButton)}
+                </Button>
+              </div>
+            </Card.Body>
+          </Card>
+        ) : (
+          <Collapsible.Advanced
+            className="collapsible-card rounded mb-3 px-3 py-2"
+            onToggle={handleToggle}
+            defaultOpen={!title}
+          >
+            <Collapsible.Trigger
+              className="collapsible-trigger d-flex border-0"
+              style={{ justifyContent: 'unset' }}
+            >
+              <Collapsible.Visible whenClosed>
+                {getHeading(false)}
+                <div className="py-2 ml-auto">
+                  <ExpandMore />
+                </div>
+              </Collapsible.Visible>
+              <Collapsible.Visible whenOpen>
+                {getHeading(true)}
+                {name !== 'General' && (
```

Comment:
Currently all topics can be deleted or renamed. The check on the default topic will be on the basis of the Id. For now we can remove this check